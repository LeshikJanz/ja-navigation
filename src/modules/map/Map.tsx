import React from "react";
import { runMap } from "./utils";

class Map extends React.Component<{ onMapInit: (result: boolean) => void }> {
  componentDidMount() {
    const result = runMap();
    this.props.onMapInit(!!result);
  }

  render() {
    return <div id="map" style={{ width: "600px", height: "400px" }} />;
  }
}

export default Map;
