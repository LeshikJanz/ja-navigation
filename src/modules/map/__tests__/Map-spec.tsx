import React from "react";
import { shallow } from "enzyme";
import Map from "../Map";

describe("Map component", () => {
  it("matches snapshot", () => {
    const onMapInit = jest.fn();
    const component = shallow(<Map onMapInit={onMapInit} />);
    expect(component).toMatchSnapshot();
  });

  it("should onMapInit to be called at componentDidMount", () => {
    const onMapInit = jest.fn();
    expect(onMapInit.mock.calls.length).toBe(0);
    const component = shallow(<Map onMapInit={onMapInit} />);
    expect(onMapInit.mock.calls.length).toBe(1);
  });
});
