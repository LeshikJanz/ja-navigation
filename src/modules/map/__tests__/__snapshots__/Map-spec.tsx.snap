// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component matches snapshot 1`] = `ShallowWrapper {}`;
